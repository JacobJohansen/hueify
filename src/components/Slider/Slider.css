input[type=range] {
  -webkit-appearance: none;
  width: 100%;
  background: transparent;
}

input[type=range]::-webkit-slider-thumb {
  -webkit-appearance: none;
}

input[type=range]:focus {
  outline: none;
}

input[type=range]::-webkit-slider-thumb {
  -webkit-appearance: none;
  height: 18px;
  width: 18px;
  border-radius: 50%;
  background: var(--white);
  cursor: pointer;
  margin-top: -8px;
  box-shadow: var(--box-shadow);
}

input[type=range]:disabled::-webkit-slider-thumb {
  display: none;
}


input[type=range]::-webkit-slider-runnable-track {
  width: 100%;
  height: 3px;
  border-right: 3px;
  cursor: pointer;
  background: var(--grey);
}

input[type=range]:focus::-webkit-slider-runnable-track {
  background: var(--blue);
}
