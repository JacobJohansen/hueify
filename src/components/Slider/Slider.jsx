import { Component } from 'preact'
import './Slider.css'

export default class Silder extends Component {
  /**
   * Slider element
   * @type {DOMElement|null}
   * @private
   */
  element = null

  /**
   * Update the element value when the defaultValue prop is changed.
   * This is a hack so the slider can slide as smooth as possible
   * @param  {Object} props
   * @private
   */
  componentWillReceiveProps ({ defaultValue }) {
    if (this.element === document.activeElement) {
      return
    }

    this.element.value = defaultValue
  }

  /**
   * Render an input element with the type of a range
   * @param {Object} props
   * @return {JSX}
   * @public
   */
  render ({ onInput, min, max, defaultValue, disabled }) {
    return (
      <input
        ref={el => (this.element = el)}
        type='range'
        min={min}
        max={max}
        disabled={disabled}
        defaultValue={defaultValue}
        onInput={onInput}
      />
    )
  }
}
