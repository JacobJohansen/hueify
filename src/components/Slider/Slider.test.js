/* eslint-env jest */
import { h, render } from 'preact'
import { default as renderToString } from 'preact-render-to-string'
import Slider from './Slider'

global.h = h

test('Should render default slider component', () => {
  const vdom = renderToString(<Slider />)
  expect(vdom).toMatchSnapshot()
})

test('Should render slider with min/max value', () => {
  const vdom = renderToString(<Slider min={0} max={100} />)
  expect(vdom).toMatchSnapshot()
})

test('Should render disabled slider', () => {
  const vdom = renderToString(<Slider disabled />)
  expect(vdom).toMatchSnapshot()
})

test('Should render with a default value', () => {
  const vdom = renderToString(<Slider defaultValue={100} />)
  expect(vdom).toMatchSnapshot()
})

test('Should trigger onInput event when slider value changes', done => {
  const onInput = () => done()

  render(
    <Slider
      onInput={onInput}
      defaultValue={0}
      min={0}
      max={100}
    />,
    document.body
  )

  const $el = document.querySelector('input')
  $el.value = 50
  $el._component.props.onInput() // TODO find out why the on input event isn't fired
})
