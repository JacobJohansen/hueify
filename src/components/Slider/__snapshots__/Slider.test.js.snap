// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render default slider component 1`] = `"<input type=\\"range\\" />"`;

exports[`Should render disabled slider 1`] = `"<input type=\\"range\\" disabled />"`;

exports[`Should render slider with min/max value 1`] = `"<input type=\\"range\\" min=\\"0\\" max=\\"100\\" />"`;

exports[`Should render with a default value 1`] = `"<input type=\\"range\\" defaultValue=\\"100\\" />"`;
